src/TypeSpec/__Private/ShapeSpec.hack

```diff
@@ -115,4 +120,21 @@ final class ShapeSpec extends TypeSpec<shape()> {
     }
     return /* HH_IGNORE_ERROR[4110] */ darray($shape);
   }
+
+  <<__Override>>
+  public function toString(): string {
+    return $this->inners
+      |> Dict\map_with_key(
+        $$,
+        ($name, $spec) ==> Str\format(
+          '  %s => %s%s',
+          $spec->toString(),
```

Comment:
Probably needs something like

```
  Str\replace($spec->toString(), "\n", "\n  ")
```

to handle indentation of nested shapes correctly.

Reply:
heh, for all of these, yep will add a test